storage/jumpbox.go

```diff
@@ -1,6 +1,10 @@
 package storage
 
-import "reflect"
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
 
 type Jumpbox struct {
 	URL       string                 `json:"url"`
```

Comment:
While a string for a URL is minimal and makes sense for a simple input, now that more logic around user/host/protocol is being added, we may want to look into using net/url.

See https://play.golang.org/p/gj0Qwq22OFB

```go
package main

import (
	"fmt"
	"log"
	"net/url"
)

func main() {
	u, err := url.Parse("ssh://some-jumpbox.com")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(u)
	if u.User == nil {
		u.User = url.User("jumpbox")
	}
	fmt.Println(u)
}
```

Reply:
An example with a bit more real SSH/Proxy example: https://play.golang.org/p/reXxFB0XTiK

Reply:
This refactor probably touches more code than this PR needs so it may be best for Infra to work it or at least x-team-pair.

Reply:
I don't see a need for or advantage of switching to `net.url` but I see a potential disadvantage as it now gets serialized/deserialized as `{"url":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""}}`

What am I missing?

Reply:
Okay, I didn't consider the serialization stuffs and agree lets keep it simple. I think this is ready to be merged without my suggested edit.

@Polar-Beard @zachgersh